agent/mitm/handlers/HttpRequestHandler.ts

```diff
@@ -257,14 +260,17 @@ export default class HttpRequestHandler extends BaseHttpHandler {
     proxyToClientResponse.writeHead(proxyToClientResponse.statusCode);
   }
 
-  private async writeResponse(): Promise<void> {
+  private async writeResponse(): Promise<number> {
     const context = this.context;
     const { serverToProxyResponse, proxyToClientResponse } = context;
 
     context.setState(ResourceState.WriteProxyToClientResponseBody);
 
+    let resourceSize = 0;
+
     for await (const chunk of serverToProxyResponse) {
       const data = context.cacheHandler.onResponseData(chunk as Buffer);
+      resourceSize += data.length;
```

Comment:
This will fail if the cache handler doesn't choose to use the body. You need to use `(chunk as Buffer).length`